walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,210 @@
 // limitations under the License.
 
 package com.google.sps;
+import com.google.sps.*;
+import java.util.*;
+public final class FindMeetingQuery {
 
-import java.util.Collection;
+  /* Given all events, return a HashMap that contains the schedule of mandatory attandees and optional attandees 
+  param: @events: all events containing time and attandees list 
+         @request: Meeting requests containing attandees list
+  return: @allSchedules a hashmap with 2 key: 'mandatory' and 'optional', the values are another HashMap whose key is attandee and 
+          value is the schedule of that attandee
+  */
+  public static HashMap<String, HashMap<String, ArrayList<TimeRange>>> getSchedules(Collection<Event> events, MeetingRequest request) {
```

Comment:
If you have a data structure this complex, it's a good signal that it might be time to make a simple class or multiple values instead. That way, the type tells you something meaningful about the data. Currently it's hard to tell what this structure means. It looks like the top-level keys are just whether it's default or mandatory, and the inner hash map is attendee-> time range. I'd recommend either taking and returning two HashMaps or creating a scheduling class that holds two HashMaps.

Reply:
Another suggestion is to potentially refactor this method into two different methods (getMandatorySchedules and getOptionalSchedules). It might seem repetitive but it might have the benefit of better readability that way.